tests/float/float_format_ints.py

```diff
@@ -0,0 +1,23 @@
+# Test that integers format to exact values.
+
+for b in [13, 123, 457, 23456]:
+    for r in range(1, 10):
+        e_fmt = "{:." + str(r) + "e}"
+        f_fmt = "{:." + str(r) + "f}"
+        g_fmt = "{:." + str(r) + "g}"
+        for e in range(0, 5):
+            f = b * (10**e)
+            title = str(b) + " x 10^" + str(e)
+            print(title, "with format", e_fmt, "gives", e_fmt.format(f))
+            print(title, "with format", f_fmt, "gives", f_fmt.format(f))
+            print(title, "with format", g_fmt, "gives", g_fmt.format(f))
+
+
+# 16777215 is 2^24 - 1, the largest integer that can be completely held
+# in a float32.
+print("{:f}".format(16777215))
+# 4294967040 = 16777215 * 128 is the largest integer that is exactly
+# represented by a float32 and that will also fit within a (signed) int32.
+# The upper bound of our integer-handling code is actually double this,
+# but that constant might cause trouble on systems using 32 bit ints.
+print("{:f}".format(2147483520))
```

Comment:
Please add the 2 test cases using the explicit representation and array.array

Reply:
Won't that fail on builds with
```
#define MICROPY_FLOAT_IMPL          (MICROPY_FLOAT_IMPL_FLOAT)
```
I guess there's some way to gate or condition tests based on build conditions.  I'll look into it.

Reply:
I'm going with a separate ```tests/float/float_format_int_doubleprec.py```, which I'll add to the ```if upy_float_precision < 64``` skips in ```run-tests.py```.

Reply:
Sounds good!